google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/CivilTimeEncoderTest.java

```diff
@@ -0,0 +1,425 @@
+        CivilTimeEncoder.encodePacked64TimeMicros(LocalTime.of(23, 59, 59, 999_000_000)));
+
+    // 00:00:00.000000
+    // 0b000000000000000000000000000|00000|000000|000000|00000000000000000000
+    // 0x0
+    assertEquals(LocalTime.of(0, 0, 0, 0), CivilTimeEncoder.decodePacked64TimeMicros(0x0L));
+    // 00:01:02.000003
+    // 0b000000000000000000000000000|00000|000001|000010|00000000000000000011
+    // 0x4200003
+    assertEquals(
+        LocalTime.of(0, 1, 2, 3_000), CivilTimeEncoder.decodePacked64TimeMicros(0x4200003L));
+    // 12:00:00.000000
+    // 0b000000000000000000000000000|01100|000000|000000|00000000000000000000
+    // 0xC00000000
+    assertEquals(
+        LocalTime.of(12, 0, 0, 0), CivilTimeEncoder.decodePacked64TimeMicros(0xC00000000L));
+    // 13:14:15.000016
+    // 0b000000000000000000000000000|01101|001110|001111|00000000000000010000
+    // 0xD38F00010
+    assertEquals(
+        LocalTime.of(13, 14, 15, 16_000), CivilTimeEncoder.decodePacked64TimeMicros(0xD38F00010L));
+    // 23:59:59.999999
+    // 0b000000000000000000000000000|10111|111011|111011|11110100001000111111
+    // 0x17EFBF423F
+    assertEquals(
+        LocalTime.of(23, 59, 59, 999_999_000),
+        CivilTimeEncoder.decodePacked64TimeMicros(0x17EFBF423FL));
+
+    try {
+      // 00:00:00.000000
+      // 0b000000000000000000000000001|00000|000000|000000|00000000000000000000
+      // 0x2000000000
+      CivilTimeEncoder.decodePacked64TimeMicros(0x2000000000L);
+      Assert.fail();
+    } catch (IllegalArgumentException e) {
+      assertEquals(null, e.getMessage());
+    }
+    try {
+      // 00:00:00.1000000
+      // 0b000000000000000000000000000|00000|000000|000000|11110100001001000000
+      // 0xF4240
+      CivilTimeEncoder.decodePacked64TimeMicros(0xF4240L);
+    } catch (IllegalArgumentException e) {
+      assertEquals(null, e.getMessage());
+    }
+    try {
+      // 00:00:60.000000
+      // 0b000000000000000000000000000|00000|000000|111100|00000000000000000000
+      // 0x3C00000
+      CivilTimeEncoder.decodePacked64TimeMicros(0x3C00000L);
+    } catch (IllegalArgumentException e) {
+      assertEquals("Invalid value for SecondOfMinute (valid values 0 - 59): 60", e.getMessage());
+    }
+    try {
+      // 24:00:00.000000
+      // 0b000000000000000000000000000|11000|000000|000000|00000000000000000000
+      // 0x1800000000
+      CivilTimeEncoder.decodePacked64TimeMicros(0x1800000000L);
+    } catch (IllegalArgumentException e) {
+      assertEquals("Invalid value for HourOfDay (valid values 0 - 23): 24", e.getMessage());
+    }
+    try {
+      // 00:60:00.000000
+      // 0b000000000000000000000000000|00000|111100|000000|00000000000000000000
+      // 0xF0000000
+      CivilTimeEncoder.decodePacked64TimeMicros(0xF0000000L);
+    } catch (IllegalArgumentException e) {
+      assertEquals("Invalid value for MinuteOfHour (valid values 0 - 59): 60", e.getMessage());
+    }
+
+    LocalTime time = LocalTime.of(1, 1, 1, 1_000_000);
```

Comment:
Put this roundtrip test with every encode test above, to make sure all encoded result can be decoded.

Reply:
As we discussed in chat, will make sure that round trip tests are present in the integration test and keep this code closer to the zetasql tests